### DISCUSSION TITLE
BUG REPORT documentation

### DISCUSSION THREAD

**Discussion by minerop5000** [elgatosf/elgato#42] [General]
Firstly:
It is not clear where to post bug reports for the https://docs.elgato.com page. Link on https://docs.elgato.com/sdk/support/help-and-bug-reports send me to this discussion page without a template for bug reports.

Now to the bug:
https://docs.elgato.com/sdk/plugins/getting-started#4.-add-the-plugin-to-stream-deck

The command for the Windows SymLink is in the wrong order. 
It needs to be `mklink /D <AppData-folder> <plugin-repo-folder>` see: https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc753194(v=ws.10)?redirectedfrom=MSDN

Also, for me, it only worked when using absolute paths for both.

**Comment by Hecsall** [comment]
Came here for the same issue, but on macOS. 
On macOS for the symlink to work it needs to be an absolute path.

For example, this will work
```sh
ln -s /Users/Username/Projects/my-plugin/src/com.example.my-plugin.sdPlugin ~/Library/Application\ Support/com.elgato.StreamDeck/Plugins/
```

While this won't (assuming you are inside your project folder)
```sh
ln -s ./src/com.example.my-plugin.sdPlugin ~/Library/Application\ Support/com.elgato.StreamDeck/Plugins/
```

_Edit_

After testing on Windows I can confirm that the `mklink` command in the documentation not only is in the wrong order but also uses the wrong variable to get the Username to build the path. At least trying on Windows 11 the %USER% variable does not exist, but %USERNAME% does.

# Solution

So to recap, the correct commands to run for the symlinks should be the ones below.
_Note that I'm assuming you run these commands from the root folder of your plugin repo._

**macOS**
```sh
# Here using $(pwd) to get the full absolute path to the plugin folder
ln -s $(pwd)/src/com.example.my-plugin.sdPlugin ~/Library/Application\ Support/com.elgato.StreamDeck/Plugins/
```

**Windows**
```powershell
# Note: this works inside the cmd, not on PowerShell
# Here using %cd% to get the full absolute path to the plugin folder
mklink /D C:\Users\%USERNAME%\AppData\Roaming\Elgato\StreamDeck\Plugins\com.example.my-plugin.sdPlugin %cd%\src\com.example.my-plugin.sdPlugin
```

**Comment by GeekyEggo** [comment]
Thanks for reporting this, and apologies for any inconvenience it's caused.

@ChekTek could you review the getting started guide to reflect the correct sym-linking please.

**Comment by ChekTek** [comment]
Thanks for pointing this out! I've updated the documentation. Shoutout to @Hecsall for providing the solution.

https://docs.elgato.com/sdk/plugins/getting-started

(you may have to clear your browser cache)
